src/cmd/ksh93/meson.build

```diff
@@ -106,6 +106,45 @@ if cc.has_function('sigqueue', prefix: '#include <signal.h>')
     shcomp_c_args += [ '-D_lib_sigqueue=1' ]
 endif
 
+
+socketpair_shutdown_mode_test = '''
+	#include <sys/types.h>
+	#include <sys/stat.h>
+	#include <sys/socket.h>
+	int main()
+	{
+		int		sfd[2];
+		struct stat	st0;
+		struct stat	st1;
+		if (socketpair(AF_UNIX, SOCK_STREAM, 0, sfd) < 0 ||
+		    shutdown(sfd[0], 1) < 0 ||
+		    shutdown(sfd[1], 0) < 0)
+			return(1);
```

Comment:
I know you just copy/pasted this code but it would be nice if some of the bogosities like the extraneous parens around the `return` expression was removed. :smile:

Reply:
:)